src/common/hook/usePublishQueue/utils/publish.tsx

```diff
@@ -1,25 +1,87 @@
-import { FormEntry, Config, RawDocument, PublishingJob } from "../../../../types";
-import { defaultsDeep, groupBy } from "lodash";
 import { wrapDocuments } from "@govtechsg/open-attestation";
+import { defaultsDeep, groupBy } from "lodash";
+import {
+  ActionsUrlObject,
+  Config,
+  DocumentStorage,
+  FormEntry,
+  PublishingJob,
+  RawDocument,
+} from "../../../../types";
+import { getQueueNumber } from "../../../API/storageAPI";
+import { encodeQrCode } from "../../../utils";
 
-export const getRawDocuments = (forms: FormEntry[], config: Config): RawDocument[] => {
-  return forms.map(({ data, templateIndex, fileName, ownership }) => {
-    const formConfig = config.forms[templateIndex];
-    if (!formConfig) throw new Error("Form definition not found");
-    const formDefaults = formConfig.defaults;
-    const formData = { ...data.formData };
-    defaultsDeep(formData, formDefaults);
-    const contractAddress =
-      formData.issuers[0]?.documentStore || formData.issuers[0]?.tokenRegistry;
-    const payload = formConfig.type === "TRANSFERABLE_RECORD" ? { ownership } : {};
-    return {
-      type: formConfig.type,
-      contractAddress,
-      rawDocument: formData,
-      fileName,
-      payload,
-    };
-  });
+const QR_CODE_OBJECT = { links: { self: { href: "" } } };
+
+interface QueueNumberTypes {
+  id: string;
+  key: string;
+}
+
+interface NetworkUrl {
+  homestead: string;
+  ropsten: string;
+  rinkeby: string;
+}
+
+const getReservedStorageUrl = async (
+  documentStorage: DocumentStorage,
+  network: "homestead" | "ropsten" | "rinkeby"
+): Promise<ActionsUrlObject> => {
+  const queueNumber = await getQueueNumber(documentStorage);
+
+  const networkUrl = {
+    homestead: "",
+    ropsten: "dev.",
+    rinkeby: "rinkeby",
+  } as NetworkUrl;
+
+  const qrUrlObj = {
+    type: "DOCUMENT",
+    payload: {
+      uri: `${documentStorage.url}/${queueNumber.data.id}`,
+      key: queueNumber.data.key,
+      permittedActions: ["STORE"],
+      redirect: `https://${networkUrl[network]}tradetrust.io/`,
```

Comment:
i think move this whole thing upwards into a const object with network->full url mappings because its not really a generalisable pattern

Reply:
ok